packages/explorer/explorer-runner.mjs

```diff
@@ -0,0 +1,110 @@
+import { watchFile } from "fs";
+import { readFile } from "fs/promises";
+import minimist from "minimist";
+import path from "path";
+import process from "process";
+import { fileURLToPath } from "url";
+import { spawn } from "child_process";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+const argv = minimist(process.argv.slice(2));
+const port = argv.port || 13690;
+const chainId = argv.chainId || 31337;
+const env = argv.env || "production";
+const indexerDbPath = argv.indexerDbPath || "indexer.db";
+const worldsConfigPath = argv.worldsConfigPath || null;
+
+let worldAddress = argv.worldAddress || null;
+let explorerProcess;
+
+async function startExplorer() {
+  let command, args;
+
+  if (env === "production") {
+    command = "pnpm";
+    args = ["start"];
+  } else {
+    command = "pnpm";
+    args = ["dev"];
+  }
+
+  explorerProcess = spawn(command, args, {
+    cwd: __dirname,
+    stdio: "inherit",
+    env: {
+      ...process.env,
+      PORT: port,
+      INIT_PWD: process.cwd(),
+      WORLD_ADDRESS: worldAddress,
+      INDEXER_DB_PATH: indexerDbPath,
+    },
+  });
+}
+
+async function readWorldsJson() {
+  try {
+    const data = await readFile(worldsConfigPath, "utf8");
+    if (data) {
+      const worlds = JSON.parse(data);
+      const world = worlds[chainId];
+      if (world) {
+        return world.address;
+      } else {
+        console.error(`World not found for chain ID ${chainId}`);
+        return null;
+      }
+    }
+  } catch (error) {
+    console.error("Error reading worlds.json:", error);
+    return null;
+  }
+}
+
+async function restartExplorer() {
+  if (explorerProcess) {
+    explorerProcess.kill();
```

Comment:
I am wondering if `worlds.json` changes the address but we don't wipe the DB, what happens?

Or if `worlds.json` changes but the address _doesn't_ change?

Reply:
Hehe all good, should think through all the scenarios and consider if it's worth handling. If `worlds.json` changes, only in case the `worldAddress` has actually changed will the process get restarted. In the context of development environment, it's probably most relevant when the `worlds.json` is initially empty, and only later the address is set so the `explorer` gets set up properly without manual intervention.

As for `indexer.db`, the `explorer` shows whatever is included in the database. So for v0 I don't think it's an issue since we are focusing on local development environment where there is only 1 database to fetch from, plus as you said it all happens on local anvil chain. But in the future versions when this actually becomes "worlds" explorer, we should consider this more carefully.